### ISSUE TITLE
[Numpy] Cannot mix numpy scalar and MXNet numpy ndarray

### ISSUE THREAD

**Issue by sxjscience** [apache/incubator-mxnet#16743]
Mixing the original numpy array and the mxnet numpy array will trigger some errors:

Minimal reproducible example:
```python
import mxnet as mx
import numpy as np
mx.npx.set_np()
a = np.array(1)
a += mx.np.array(1)
```

Error:
```
AssertionError                            Traceback (most recent call last)
<ipython-input-6-4f0c781027dc> in <module>
      4 mx.npx.set_np()
      5 a = np.array(1)
----> 6 a += mx.np.array(1)

~/mxnet/python/mxnet/numpy/multiarray.py in __array_ufunc__(self, ufunc, method, *inputs, **kwargs)
    209                                  'to the list.'
    210                                  .format(name))
--> 211             return mx_ufunc(*inputs, **kwargs)
    212         else:
    213             return NotImplemented

~/mxnet/python/mxnet/util.py in _wrap_np_binary_func(x1, x2, out, **kwargs)
    662                     # otherwise raise TypeError with not understood error message
    663                     raise TypeError("{} {} not understood".format(key, value))
--> 664         return func(x1, x2, out=out)
    665     return _wrap_np_binary_func
    666 

~/mxnet/python/mxnet/numpy/multiarray.py in add(x1, x2, out, **kwargs)
   2415            [ 6.,  8., 10.]])
   2416     """
-> 2417     return _mx_nd_np.add(x1, x2, out)
   2418 
   2419 

~/mxnet/python/mxnet/util.py in _wrap_np_binary_func(x1, x2, out, **kwargs)
    662                     # otherwise raise TypeError with not understood error message
    663                     raise TypeError("{} {} not understood".format(key, value))
--> 664         return func(x1, x2, out=out)
    665     return _wrap_np_binary_func
    666 

~/mxnet/python/mxnet/ndarray/numpy/_op.py in add(x1, x2, out, **kwargs)
    525         The sum of x1 and x2, element-wise. This is a scalar if both x1 and x2 are scalars.
    526     """
--> 527     return _ufunc_helper(x1, x2, _npi.add, _np.add, _npi.add_scalar, None, out)
    528 
    529 

~/mxnet/python/mxnet/ndarray/numpy/_op.py in _ufunc_helper(lhs, rhs, fn_array, fn_scalar, lfn_scalar, rfn_scalar, out)
    389         return lfn_scalar(lhs, float(rhs), out=out)
    390     elif isinstance(rhs, ndarray):
--> 391         return fn_array(lhs, rhs, out=out)
    392     else:
    393         raise TypeError('type {} not supported'.format(str(type(rhs))))

~/mxnet/python/mxnet/ndarray/register.py in add(lhs, rhs, out, name, **kwargs)

AssertionError: Argument lhs must have NDArray type, but got 1
```

**Comment by sxjscience** [comment]
This will be fine:
```python
import mxnet as mx
import numpy as np
mx.npx.set_np()
a = 1
a += mx.np.array(1)
```

**Comment by samskalicky** [comment]
@zachgk assign @sxjscience

**Comment by yzhliu** [comment]
@BenjaminCHEN2016 will be looking into it.

**Comment by BenjaminCHEN2016** [comment]
@sxjscience @yzhliu Are numpy array and mxnet array suppose to work together ?

**Comment by sxjscience** [comment]
@BenjaminCHEN2016 The problem is that it's a numpy scalar and should be treated similarly as python scalars. I just checked that Jax supports the following:
```python
import jax.numpy as np
import numpy as onp
a = onp.array(1)
b = np.array(1)
a
```
Output:
```
Out[17]: DeviceArray(2, dtype=int32)
```

**Comment by BenjaminCHEN2016** [comment]
@yzhliu Issue resolved in #18313

**Comment by yzhliu** [comment]
closed by #18313
